### ISSUE TITLE
Exon end is larger than chromosome

### ISSUE THREAD

**Issue by mtekman** [alexdobin/STAR#988]
Running on `2.7.5a` or `2.7.5b`:

```bash
    STAR --runMode genomeGenerate \
         --genomeDir tempstar \
         --readFilesCommand zcat \
         --genomeFastaFiles $FASTA \
         --sjdbGTFfile $GTF \
         --sjdbOverhang 100 \
         --genomeSAindexNbases 4 \
         --runThreadN 5
```

The warning message it gives is:

```
  : Aug 03 23:34:47 ..... started STAR run
  : Aug 03 23:34:47 ... starting to generate Genome files
  : Aug 03 23:34:47 ..... processing annotations GTF
  : !!!!! WARNING: while processing sjdbGTFfile=filtered2.Homo_sapiens.GRCh38.100.chr21.gtf, line:
  : 21	havana	exon	5011799	5011874	.	+	.	gene_id "ENSG00000279493"; gene_version "1"; transcript_id "ENST00000624081"; transcript_version "1"; exon_number "1"; gene_name "FP565260.4"; gene_source "havana"; gene_biotype "protein_coding"; transcript_name "FP565260.4-201"; transcript_source "havana"; transcript_biotype "protein_coding"; exon_id "ENSE00003760288"; exon_version "1"; tag "cds_start_NF"; tag "mRNA_start_NF"; tag "basic"; transcript_support_level "5";
  :  exon end = 5011874 is larger than the chromosome 21 length = 697679 , will skip this exon
```

but it actually hangs forever with 0% CPU until I kill it.

#### Debugging the Error Message

The header for this particular chromosome file is adapted from the one from Ensembl:
 * ftp://ftp.ensembl.org/pub/release-100/fasta/homo_sapiens/dna/

```
>21 dna:chromosome chromosome:GRCh38:21:5011741:5709420:1 REF
CCCTGGCCTGTGGGCCCTGCGAGCTGAGCAGGGTCTGCCCTGGGGTCATCTCGTGCTGCC
GCACGCGACCTTGCCCACTGCTGCTGCCTCTCAGTGCCCACTCCCCTCTGTCCTGCAGGT
```

meaning it is only `697,679` bp long but it starts at position `5,011,741`.

STAR seems to ignore this part of the header and assumes the chromosome starts at position `1`.

#### Files for reproducibility

(Prefix: "filtered2")

[filtered2.Homo_sapiens.GRCh38.dna.chromosome.21.fa.txt](https://github.com/alexdobin/STAR/files/5022023/filtered2.Homo_sapiens.GRCh38.dna.chromosome.21.fa.txt)
[filtered2.Homo_sapiens.GRCh38.100.chr21.gtf.txt](https://github.com/alexdobin/STAR/files/5022025/filtered2.Homo_sapiens.GRCh38.100.chr21.gtf.txt)

# Workaround

I can get around this by subtracting `5010799` (to ensure the first exon starts at position 1000 (for stability)) from all positions in the FASTA header and the all positions in the GTF file. After that it generates the index fine.

```bash
     cat filtered2.Homo_sapiens.GRCh38.100.chr21.gtf \
       | awk -F $'\t' 'BEGIN {OFS = FS} /#!/ {print;} /21/ {$4=($4-5010799);$5=($5-5010799); print}' \
       > filtered3.Homo_sapiens.GRCh38.100.chr21.gtf

     cat filtered2.Homo_sapiens.GRCh38.dna.chromosome.21.fa \
       | awk -F ':' 'BEGIN {OFS = FS} NR==1 {$5=($5-5010799);$6=($6-5010799);print} NR>1 {print;}' \
       > filtered3.Homo_sapiens.GRCh38.dna.chromosome.21.fa
```

#### Files for reproducibility

(Prefix: "filtered3")

[filtered3.Homo_sapiens.GRCh38.100.chr21.gtf.txt](https://github.com/alexdobin/STAR/files/5022027/filtered3.Homo_sapiens.GRCh38.100.chr21.gtf.txt)
[filtered3.Homo_sapiens.GRCh38.dna.chromosome.21.fa.txt](https://github.com/alexdobin/STAR/files/5022028/filtered3.Homo_sapiens.GRCh38.dna.chromosome.21.fa.txt)

**Comment by alexdobin** [comment]
Hi Mehmet,

it's not supposed to hang after the warning, I will need to check that.

But you need to convert the GTF coordinates, or, alternatively, fill the chromosome sequence with N until position 5011741. I think the `chromosome:GRCh38:21:5011741:5709420:1` formatting is idiosyncratic for ENSEMBL. STAR actually does not read past the first blank in the chromosome name line.

Cheers
Alex

**Comment by mtekman** [comment]
Hi Alex,

I suspected it might have been an Ensembl-only thing but wasn't sure. In the end I did convert the GTF coordinates (and the FASTA too, though I guess I didn't need to). I am slightly against the idea of padding a 700 kbp sequence with 50Mbp of N's... but I guess if it's gzipped it won't add too much.

Thanks again!

**Comment by alexdobin** [comment]
Hi Mehmet,

It may seem wasteful to have 50Mb of Ns, but might be easier than converting coordinates - especially if you need alignments in the normal chromosome coordinates. Ns do not slow down the mapping as they are not indexed.

Cheers
Alex

**Comment by alexdobin** [comment]
Hi Mehmet,

The issue with hanging after the warning is fixed in 2.7.5c.
Thanks for reporting it!

Cheers
Alex
